### ISSUE TITLE
next year solution to opt out when interact

### ISSUE THREAD

**Issue by ScottDowne** [MozillaFoundation/Advocacy#690]
Doubtful we have time to do this for 2016. Wish I thought of it sooner.

Anyway, the idea is pretty simple. If they click donate in the snippet, we store a separate key in indexdb that's not the same as the opt out one.

Next time they see the snippet, we check for this key, and if it's there, we present a nice little thank you or something, along with a back button in case they want to give again, instead of the donate snippet.

We have to be a little careful about the thank you message, as there is no guarantee they have given yet, just that they have clicked, so we don't want to confuse people into thinking they have given, when they have not.

I don't think we should put in a signup or share cta here as if they have donated, they have likely already seen that, but maybe?

It's really not a huge amount of work, and with quite a lot of potential.

cc @Matt-Holland @HPaulJohnson @lovegushwa @roseregina @beccaklam @natalieworth

**Comment by roseregina** [comment]
In theory I like this, but I think that if we go this route, we should figure out how to include a  "why am I seeing this?" explainer link on the "thank you" snippet to help reduce the number of people who are freaked out by it because they didn't give and think that this means that we took money from them because they clicked on the donate button even though they didn't put in any payment information. 

We may also want to specifically do wording that is more like "it seems like you may have already donated. Thank you! if you didn't and you would like to, please click here." so that people who click on the donate button but do not complete the donation are able to easily find were to donate again and don't think that this means that they already did and they just don't remember. 

Stray thoughts that are probably not relevant to this discussion: 
* how is this different than a cookie? 
* will it upset a portion of our user base that we are "tracking" them (which we aren't, but probably only to seriously pedantic people like me)? 
* how does this mesh with EU regs regarding cookies and would we have to add that stupid disclaimer bar say that by being here you agree to accept? if we would need to put that kind of thing in, I think this is not really going to work, but I also think there's a really good chance that legally we wouldn't have to, but INAL;TINLA. 

TL;DR: is this going to increase false positives with "missed" donors? how should we be transparent about this? are there EU regulator concerns?

**Comment by ScottDowne** [comment]
Snippets already do a great deal using indexeddb and some of its usage is
baked directly into the browser. The snippet itself is loading into
indexeddb. Opt out uses indexeddb currently too. It's not sent to the
server like a cookie and about:home is a very different kind of webpage.
I'm not super worried about actual security or privacy concerns. But we
should have a ready made response for this.

I'm more worried about our wording of the message. As we both pointed out
there is no way to send a message to the snippet after a donation. So the
message can not say anything about making a donation. We cannot assume
that. But I'm sure there is an opportunity here to do something better than
asking for more money or disabling the snippet.

**Comment by roseregina** [comment]
Thanks for the clarification on indexeddb versus a cookie, @ScottDowne, and I do think you are right that this isn't a real issue for security or privacy, just that it may be perceived as such. 

Wording will be tricky, but I do *really* like the idea of a second chance snippet.

**Comment by ScottDowne** [comment]
I like how you coined it a "second chance snippet" I think that's an
appropriate lense to view the problem.

**Comment by roseregina** [comment]
case 20207

> Dear Mozilla,
> 
> I work in IP myself and regularly donate to open source projects. For instance I have already donated twice to Mozilla this Christmas. I think you are doing a great job and you are worth the credit.
> 
> However, just having donated, the begging screen comes back and stares me in the face when I turn my browser on the next time. This is really demotivating, like when you just gave a beggar some money in the subway and the next one just enters your train. Gives you a feeling there is no end to it.
> 
> For those of us who actually donate, why do you not provide a solution to feed the donation info back to the browser to show appreciation of the donor? Or, if that is to complicated, why not just provide a button „I have already donated“ to click? It would feel so much better than having the browser asking you for money just hours of your last donation :-(
> 
> And - to anticipate you answer, YES, the people who do not donate may click on the same button to get rid of the beggar screen. So what? Anyone can open a new window as well so nothing is lost or changed except that the non-donors get a bad feeling of clicking without actually having contributed.

more information on the problem donor was having from a later message: 
> The problem I had (and still have) is probably a glitch that will occur rarely so I do think you should spend any time on it. The machine is a MAC Pro that was upgraded from El Capitan (10.11.6) to Sierra (10.12.1) some weeks ago and since then several add-ons in FF have stopped working so I will make a clean install and hopefully these things go away.
> 
> On two other machines it worked exactly as you said, one with Sierra (10.12.1) the other with El Capitan (10.11.6) running 50.1.0 of FF. So the problem most likely only happened to me due to the intermediate state.
> 
> Thanks for responding and keep up the good work.
> 
> PS I already tried what you wanted me to do several times before I sent the mail, it does not change anything. Also rebooting does not help.

**Comment by ScottDowne** [comment]
I suspect they saw this again when we turned on the spectacular snippet, which is considered a new snippet.

Or they donated before we added the feature to turn it off after a donation, which is only a few weeks old.

Or we have a legit bug.

But it's a good example that doing nothing here is probably bad.

For what it's worth, we are losing about 3% of donations because of this opt out on donate feature.

@roseregina Have you responded, and if so, what did you say?

**Comment by roseregina** [comment]
I responded with a "thanks for the feedback, we've been talking about this part of the donor experience and it was good to get your input" kind of response. 

my guess is that he probably emailed because he clicked it off after donating and then it came back up with the spectacular snippet. 

I put the comment in more for the idea of an "already donated button" and to show donor support for the already donated thank you/ second chance snippet idea.

**Comment by ScottDowne** [comment]
Ah, makes perfect sense. Thanks! 🚀
